src/main/java/uk/gov/companieshouse/insolvency/delta/mapper/CaseMapper.java

```diff
@@ -65,11 +69,25 @@ default void mapDatesAndLinks(@MappingTarget ModelCase modelCase,
 
         Optional<Long> optionalMortgageId = Optional.ofNullable(sourceCase.getMortgageId());
         optionalMortgageId.ifPresent(mortgageId -> {
-            String link = String.format("/company/%s/charges/%s", companyNumber, mortgageId);
+            try {
+                // Concatenate mortgageId with salt, hash using SHA-1 and then encode using base64
+                MessageDigest digest = MessageDigest.getInstance("SHA-1");
+                var salt = "jkdhjdio";
+                String saltedMortgageId = mortgageId + salt;
+                byte[] hash = digest.digest(saltedMortgageId.getBytes(StandardCharsets.UTF_8));
+                // We remove the padding equals '=' appended at the end by the base64 encoding
+                // (could be at most 2 = based on the length of the mortgage id).
+                var encodedMortgageId = Base64Utils.encodeToUrlSafeString(hash).replace("=", "");
```

Comment:
probably can use encodeBase64URLSafeString instead of replacing "=" manually?

Reply:
Good shout, done and tested.